oryx-app-mllib/src/main/java/com/cloudera/oryx/app/mllib/kmeans/KMeansEvaluation.java

```diff
@@ -122,10 +125,54 @@
 
     double dunnIndex = minInterClusterDistance / maxIntraClusterDistance;
     log.info("Computed Dunn Index for {} clusters: {}", numClusters, dunnIndex);
-
     return dunnIndex;
   }
 
+  /**
+   * Computes the Silhouette Coefficient for this clustering, range is [-1,1]
+   * (http://www.sciencedirect.com/science/article/pii/0377042787901257)
+   * Here we are computing the overall Silhouette Coefficient across all of the clustering
+   * Another option is to compute the Silhouette coefficient for each of the individual clusters
+   * @param evalData data for evaluation
+   * @return Silhouette Coefficient, closer to 1 is better
+   */
+  double silhouetteCoefficient(JavaRDD<Vector> evalData) {
+    JavaRDD<Vector> data = evalData;
+
+    if (evalData.count() > MAX_SAMPLE_SIZE) {
+      data = evalData.sample(false, 1.0 * MAX_SAMPLE_SIZE / evalData.count(),
```

Comment:
I'd cast with `(double)` instead. Also I don't think you need to pass a seed if you just want a random seed.